hacking/target-clock-generation.md

```diff
@@ -0,0 +1,144 @@
+When it does this, it sets TMS high, sets the jtag_proc structure back up, and then [runs a >50 pulse JTAG
+reset and the ARM SWD-to-JTAG sequence](https://github.com/blackmagic-debug/blackmagic/blob/799a4088e6c98fcbd977d9c3f2036bef4ba1e9b6/src/platforms/common/jtagtap.c#L53-L55).
+
+This then ensures the bus is in a good state while leaving TMS high after. We then perform an additional
+JTAG reset when starting the ID code readout, along with having to switch into the Shift-DR JTAG TAP state. These
+are the additional sets of pulses between the marked ones of interest. We finally do
+[several 32-bit transactions](https://github.com/blackmagic-debug/blackmagic/blob/799a4088e6c98fcbd977d9c3f2036bef4ba1e9b6/src/target/jtag_scan.c#L171)
+on the bus, with the goal of reading out ID codes. These are done with TMS low giving this distinctive signature
+between TMS and TCK that we see in the capture.
+
+### Making the measurements
+
+We make multiple measurements in each of the following steps due to the issue of jitter caused by the capture
+temporal resolution resulting in quantisation noise. That is, if we measure only one cycle, the value measured
+may be off by a substantial enough factor that it introduces excessive error into subsequent calculations.
+We counter that by making several measurements of each section of pulses, and applying a standard average as this
+compensates for this capture jitter, improving the method's accuracy.
+
+![A detailed view of the first pulse section of interest](../_assets/target-clk-gen/first_pulse_train.png)
+
+We start by zooming in on the first pulse section of interest, and then using measurement markers to make
+several (at least 10) single-period measurements of the clock pulses generated by jtagtap_next(), making
+notes of the results and how many of each value are observed. Add the result up and divide by the total number
+of samples taken. Record the final result for later use.
+
+We then repeat for jtagtap_tms_seq(), making a record of the final result using the same multiple measurement
+and averaging method.
+
+![A detailed view of the second pulse section of interst](../_assets/target-clk-gen/second_pulse_train.png)
+
+Finally, we zoom back out, pan over and then zoomed back in on the second section of interest. We then repeat
+the measurement protocol on this final set of pulses and make a record of the final result after averaging.
+
+### Deducing the final TCK clock frequency
+
+After following the measurement protocol outlined above, you will now have 3 values - a clock frequency achieved
+by jtagtap_next(), jtagtap_tms_seq() and jtagtap_tdi_tdo_seq().
+
+It is tempting to take the fastest of these and say "that's the clock frequency!", but this would be misleading
+to the user and especially when trying to make comparative measurements to other probes and the clock frequencies
+they achieve. So instead, for comparative accuracy, we must average between these 3 measurements.
```

Comment:
Nitpick: I'm not sure if you meant here that you intend to compare probes' frequencies (comparative measurements) or, that you want to take measurements with a reproducible setup of different probes (comparable measurements).

Reply:
Technically, both. The direct intended meaning is comparing between different probes' frequencies, hence "comparative measurements" - please let us know if you think we should alter this wording or keep it the same.

Reply:
*nod* Keep the former, please. Marking as fixed.